Patch: 70_invalid_utf8_dos.patch
Author: Lionel Elie Mamane <lionel@mamane.lu>
Avoid DOS if attachement filename is invalid Unicode string (Closes: #327732, CVE-2005-3573)
Index: b/Mailman/Handlers/Scrubber.py
===================================================================
--- a/Mailman/Handlers/Scrubber.py	2010-05-24 17:50:44.000000000 +0200
+++ b/Mailman/Handlers/Scrubber.py	2010-05-24 17:51:12.000000000 +0200
@@ -300,7 +300,10 @@
                 os.umask(omask)
             desc = part.get('content-description', _('not available'))
             desc = Utils.oneline(desc, lcset)
-            filename = part.get_filename(_('not available'))
+            try:
+                filename = part.get_filename(_('not available'))
+            except UnicodeDecodeError:
+                filename = _('not available')
             filename = Utils.oneline(filename, lcset)
             replace_payload_by_text(part, _("""\
 A non-text attachment was scrubbed...
@@ -422,7 +425,10 @@
     ctype = msg.get_content_type()
     # i18n file name is encoded
     lcset = Utils.GetCharSet(mlist.preferred_language)
-    filename = Utils.oneline(msg.get_filename(''), lcset)
+    try:
+        filename = Utils.oneline(msg.get_filename(''), lcset)
+    except UnicodeDecodeError:
+        filename = ''
     filename, fnext = os.path.splitext(filename)
     # For safety, we should confirm this is valid ext for content-type
     # but we can use fnext if we introduce fnext filtering
